### PR TITLE
Kernel: Add barebones ACPI AML parsing support

### PR DESCRIPTION
**Pull request by tomuta**
_This is *very* early and incomplete, so don't waste your time on reviewing the code at this point, it will likely change significantly!_

- [ ] Get the included test.aml to run in the unit test

### CODE REVIEW

File: Kernel/ACPI/Parser.h [hunk 1]
```diff
@@ -90,6 +97,8 @@ class Parser {
     Vector<PhysicalAddress> m_sdt_pointers;
     PhysicalAddress m_fadt;
     PhysicalAddress m_facs;
+    u64 m_dsdt { 0 };
```
Comment (supercomputer7) [inline Kernel/ACPI/Parser.h:100]: Is `m_dsdt` supposed to indicate the physical address of the the DSDT table? if so, why not to use `PhysicalAddress` instead?
Also, remember that the FADT holds this value twice (one for ACPI 1, and an extended pointer for later revisions).

**Comment by supercomputer7** [comment]
I also wanted to mention the decision of making a `LibAML`, which is imho a very good one. Maybe we can write a userspace parser, so we could compile ASL files into AML blobs, but that's a feature that we will think about for a long time to come  :)

File: Kernel/ACPI/Parser.h (hunk 1 above)
Comment (tomuta) [reply Kernel/ACPI/Parser.h:100]: Yeah eventually. The problem is that you can have 64 bit physical addresses on a 32 bit machine, but we use `FlatPtr` only. In fact, there's two DSDT pointers, an optional 64 bit physical address and the legacy 32 bit physical address.

**Comment by tomuta** [comment]
Writing a "decompiler" with this should be trivial, it's essentially just dumping each node as a string...

File: Kernel/ACPI/Parser.h (hunk 1 above)
Comment (supercomputer7) [reply Kernel/ACPI/Parser.h:100]: Then I think we should make `PhysicalAddress` to wrap a `u64` value as it represents a physical address which can be as you said 64 bits on 32 bit machine.
About the DSDT pointer, AFAIK most machines use both DSDT pointers to point to the same physical address as it doesn't make sense to make two copies of the same data. On my machine I decompiled the FADT blob and saw that this is true.

**Pull request closed without merging**